x-pack/plugins/endpoint/public/embeddables/resolver/store/middleware.ts

```diff
@@ -19,22 +20,61 @@ export const resolverMiddlewareFactory: MiddlewareFactory = context => {
     next(action);
     if (action.type === 'userChangedSelectedEvent') {
-      if (context?.services.http) {
+      if (context?.services.http && action.payload.selectedEvent) {
         api.dispatch({ type: 'appRequestedResolverData' });
-        const uniquePid = action.payload.selectedEvent?.endgame?.unique_pid;
-        const legacyEndpointID = action.payload.selectedEvent?.agent?.id;
-        const [{ lifecycle }, { children }, { events: relatedEvents }] = await Promise.all([
-          context.services.http.get(`/api/endpoint/resolver/${uniquePid}`, {
-            query: { legacyEndpointID },
-          }),
-          context.services.http.get(`/api/endpoint/resolver/${uniquePid}/children`, {
-            query: { legacyEndpointID },
-          }),
-          context.services.http.get(`/api/endpoint/resolver/${uniquePid}/related`, {
-            query: { legacyEndpointID },
-          }),
-        ]);
-        const response = [...lifecycle, ...children, ...relatedEvents];
+        let response = [];
+        let lifecycle: ResolverEvent[];
+        let childEvents: ResolverEvent[];
+        let relatedEvents: ResolverEvent[];
+        const ancestors: ResolverEvent[] = [];
+        const maxAncestors = 5;
+        if (isLegacyEvent(action.payload.selectedEvent)) {
+          const uniquePid = action.payload.selectedEvent?.endgame?.unique_pid;
+          const legacyEndpointID = action.payload.selectedEvent?.agent?.id;
+          [
+            { lifecycle },
+            {
+              children: [{ lifecycle: childEvents }],
+            },
+            { events: relatedEvents },
+          ] = await Promise.all([
+            context.services.http.get(`/api/endpoint/resolver/${uniquePid}`, {
+              query: { legacyEndpointID },
+            }),
+            context.services.http.get(`/api/endpoint/resolver/${uniquePid}/children`, {
+              query: { legacyEndpointID },
+            }),
+            context.services.http.get(`/api/endpoint/resolver/${uniquePid}/related`, {
+              query: { legacyEndpointID },
+            }),
+          ]);
+        } else {
+          const uniquePid = action.payload.selectedEvent.endpoint.process.entity_id;
+          const ppid = action.payload.selectedEvent.endpoint.process.parent?.entity_id;
+          async function getAncestors(pid: string | undefined) {
+            if (ancestors.length < maxAncestors && pid !== undefined) {
+              const parent = await context?.services.http.get(`/api/endpoint/resolver/${pid}`);
+              ancestors.push(parent.lifecycle[0]);
+              if (parent.lifecycle[0].endpoint?.process?.parent?.entity_id) {
+                await getAncestors(parent.lifecycle[0].endpoint.process.parent.entity_id);
+              }
+            }
+          }
+          [
+            { lifecycle },
+            {
+              children: [{ lifecycle: childEvents }],
+            },
+            { events: relatedEvents },
```

Comment:
❔ Maybe also receive ancestors here instead of setting it above in the `let` and pushing to it